api/rest/restapi.go

```diff
@@ -376,7 +388,7 @@ func (api *API) routes() []route {
 		{
 			"PinPath",
 			"POST",
-			"/pins/{keyType:ipfs|ipns|ipld}/{path:.*}",
+			"/pins/{path:.*}",
```

Comment:
If we leave it like that we won't be able to support `<cid>/path/to/file` format of paths. I can live without not supporting that.

But after that we will surely need to add the logic back for getting a valid path, by attaching keytype.

Reply:
> If we leave it like that we won't be able to support `<cid>/path/to/file` format of paths. I can live without not supporting that.

Yes, that's what I already said here https://github.com/ipfs/ipfs-cluster/pull/634#issuecomment-457554763.


> currently in this PR, cmd uses only PinPath( because it can do everything that Pin can do), once we change this, we can't. We will have to use Pin() as well.

No, you just need to parse the user given path (https://github.com/ipfs/go-path/blob/master/path.go#L97) so that you always send a /ipfs/<cid> when the user only provided a <cid>